PKHeX/PKM/PK3.cs

```diff
@@ -116,7 +116,7 @@ public PK3(byte[] decryptedData = null, string ident = null)
         public override int IV_SPA { get { return (int)(IV32 >> 20) & 0x1F; } set { IV32 = (uint)((IV32 & ~(0x1F << 20)) | (uint)((value > 31 ? 31 : value) << 20)); } }
         public override int IV_SPD { get { return (int)(IV32 >> 25) & 0x1F; } set { IV32 = (uint)((IV32 & ~(0x1F << 25)) | (uint)((value > 31 ? 31 : value) << 25)); } }
         public override bool IsEgg { get { return ((IV32 >> 30) & 1) == 1; } set { IV32 = (uint)((IV32 & ~0x40000000) | (uint)(value ? 0x40000000 : 0)); } }
-        public override int AbilityNumber { get { return 1 << (int)((IV32 >> 31) & 1); } set { IV32 = (IV32 & 0x7FFFFFFF) | (value > 1 ? 0x80000000 : 0); } }
+        public byte AbilityBit { get { return (byte)(IV32 >> 31); } set { IV32 = (IV32 & 0x7FFFFFFF) | (uint)(value << 31); } }
```

Comment:
This property is a single bit, right?  Why not make it a boolean?
```
public bool AbilityBit { get { return (IV32 >> 31) == 1; } set { IV32 = (IV32 & 0x7FFFFFFF) | (uint)(value ? 1 : 0 << 31); } }
```
Line 32-33 of MainPK3.cs would then become the following:
```
bool abil = pk3.AbilityBit;
CB_Ability.SelectedIndex = abil && CB_Ability.Items.Count == 0 ? 0 : 1;
```